src/main/java/org/broadinstitute/hellbender/utils/SequenceDictionaryUtils.java

```diff
@@ -0,0 +1,492 @@
+     *
+     * @param name1 name associated with dict1
+     * @param dict1 the sequence dictionary dict1
+     * @param name2 name associated with dict2
+     * @param dict2 the sequence dictionary dict2
+     * @param isReadsToReferenceComparison true if one of the dictionaries comes from a reads data source (eg., a BAM),
+     *                                     and the other from a reference data source
+     * @param intervals the user-specified genomic intervals: only required when isReadsToReferenceComparison is true,
+     *                  otherwise can be null
+     */
+    public static void validateDictionaries( final String name1,
+                                             final SAMSequenceDictionary dict1,
+                                             final String name2,
+                                             final SAMSequenceDictionary dict2,
+                                             final boolean isReadsToReferenceComparison,
+                                             final GenomeLocSortedSet intervals ) {
+
+        final SequenceDictionaryCompatibility type = compareDictionaries(dict1, dict2);
+
+        switch ( type ) {
+            case IDENTICAL:
+                return;
+            case COMMON_SUBSET:
+                return;
+            case NO_COMMON_CONTIGS:
+                throw new UserException.IncompatibleSequenceDictionaries("No overlapping contigs found", name1, dict1, name2, dict2);
+
+            case UNEQUAL_COMMON_CONTIGS: {
+                List<SAMSequenceRecord> x = findDisequalCommonContigs(getCommonContigsByName(dict1, dict2), dict1, dict2);
+                SAMSequenceRecord elt1 = x.get(0);
+                SAMSequenceRecord elt2 = x.get(1);
+
+                // todo -- replace with toString when SAMSequenceRecord has a nice toString routine
+                UserException ex = new UserException.IncompatibleSequenceDictionaries(String.format("Found contigs with the same name but different lengths:\n  contig %s = %s / %d\n  contig %s = %s / %d",
+                        name1, elt1.getSequenceName(), elt1.getSequenceLength(),
+                        name2, elt2.getSequenceName(), elt2.getSequenceLength()),
+                        name1, dict1, name2, dict2);
+
+                throw ex;
+            }
+
+            case NON_CANONICAL_HUMAN_ORDER: {
+                UserException ex;
+                if ( nonCanonicalHumanContigOrder(dict1) ) {
+                    ex = new UserException.LexicographicallySortedSequenceDictionary(name1, dict1);
+                }
+                else {
+                    ex = new UserException.LexicographicallySortedSequenceDictionary(name2, dict2);
+                }
+
+                throw ex;
+            }
+
+            case OUT_OF_ORDER: {
+                UserException ex = new UserException.IncompatibleSequenceDictionaries(
+                        "The contig order in " + name1 + " and " + name2 + "is not "
```

Comment:
Error message should be more precise: "The relative ordering of the common contigs in " + name1 + " and " + name2 + " is not the same....etc."